src/models/model.rs

```diff
@@ -19,27 +14,28 @@ use crate::input_modeling::uniform_rng::UniformRNG;
 use crate::utils::error::SimulationError;
 
 /// The type of the Model.
-#[derive(Serialize, Clone, Copy)]
+#[enum_dispatch(Model)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "type")]
 pub enum Type {
-    ExclusiveGateway,
-    Gate,
-    Generator,
-    LoadBalancer,
-    ParallelGateway,
-    Processor,
-    StochasticGate,
-    Storage,
+    ExclusiveGateway(ExclusiveGateway),
```

Comment:
Is it possible to avoid the name wrapping and duplication here, with some alternative syntax or approach?  It's not an exact comparison, but the [`ModelType` addition](https://github.com/ndebuhr/sim/pull/2/files#diff-2475200edc0ec117ad9f22f61f7374ffda697d5488639689a5801ea729cc4962R65) proposed in https://github.com/ndebuhr/sim/pull/2 avoids the wrapping.

Reply:
that's the price to pay for automatic `type` field braching on serialization, #2 deals with this problem thanks to that part of code: https://github.com/ndebuhr/sim/pull/2/files#diff-2475200edc0ec117ad9f22f61f7374ffda697d5488639689a5801ea729cc4962R19-R23

the way it works is: 
1. if we'd have
```rust
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "type")]
enum Model {
    VariantName(StructName)
}
```
2. we could use it to deserialize following yaml
```yaml
- someStructProperty1: "rast"
  someStructProperty2: "rast"
```

Reply:
I'm reasonably sure the use of `#[enum_dispatch(Model)]` itself allows you to skip the enum variants' fields (as long as the name of the enum variant and the type inside are the same). The `enum_dispatch` proc macro will generate the necessary boilerplate.

Reply:
no way, wow. thanks!